File: lantoli/spec/primefactors_spec.rb
require "rspec"
require 'primefactors.rb'

describe PrimeFactors do

  primeTests = {
    2 => [2],
    3 => [3],
    6 => [2,3],
    4 => [2,2],
    2*3*5*7*11*13 => [2,3,5,7,11,13],
    288 => [2,2,2,2,2,3,3],
    24 => [2,2,2,3],
    100 => [2,2,5,5],
    123456789 => [3,3,3607,3803],
    123456791 => [123456791],
    1234567927 => [1234567927],
    2469135854 => [2,1234567927]
  }

  primeTests.each do | number, factors|
    it "should return #{factors} for #{number}" do
       PrimeFactors.generate(number).should eq( factors )
     end
  end


end

File: josmas/jasmine-JS/spec/PrimeFactorsSpec.js
describe("Prime Factors", function() {

  it("should return an empty array for the number 1", function() {
    expect([]).toEqual(PRIMEFACTORS.generate(1));
  });

  it("should return an array with the number 2 for the number 2", function() {
    expect([2]).toEqual(PRIMEFACTORS.generate(2));
  });

  it("should return an array with the number 3 for the number 3", function() {
    expect([3]).toEqual(PRIMEFACTORS.generate(3));
  });

  it("should return an array with the numbers 2,2 for the number 4", function() {
    expect([2,2]).toEqual(PRIMEFACTORS.generate(4));
  });

  it("should return an array with the numbers 2,3 for the number 6", function() {
    expect([2,3]).toEqual(PRIMEFACTORS.generate(6));
  });

  it("should return an array with the numbers 2,2,2 for the number 8", function() {
    expect([2,2,2]).toEqual(PRIMEFACTORS.generate(8));
  });

  it("should return an array with the numbers 3,3 for the number 9", function() {
    expect([3,3]).toEqual(PRIMEFACTORS.generate(9));
  });
});

var PRIMEFACTORS = (function () {
  var primefactors = {};

  primefactors.generate = function (number){
    var result = [];
    for (var candidate = 2; number > 1; candidate += 1)
      for (; number % candidate === 0; number /= candidate)
        result.push(candidate);

    return result;
  };

  return primefactors;
}());
